Source/SCM.SwissArmyKnife/Compression/Gzip.cs

```diff
@@ -0,0 +1,79 @@
+using System.IO;
+using System.IO.Compression;
+using System.Text;
+
+namespace SCM.SwissArmyKnife.Compression
+{
+    /// <summary>
+    /// Various utility functions for working with Gzip.
+    /// </summary>
+    public static class Gzip
+    {
+        /// <summary>
+        /// This method will compress a given byte array <paramref name="dataToCompress"/> using gzip.
+        /// </summary>
+        /// <param name="dataToCompress">The byte array that will be compressed.</param>
+        /// <returns>The compressed byte array.</returns>
+        public static byte[] Compress(byte[] dataToCompress)
+        {
+            using var memoryStream = new MemoryStream();
+            using var gzipStream = new GZipStream(memoryStream, CompressionMode.Compress, true);
+
+            gzipStream.Write(dataToCompress, 0, dataToCompress.Length);
+
+            gzipStream.Dispose();
+            memoryStream.Dispose();
```

Comment:
Do you have to dispose them manually when wrapping them in `using` statements? I think it should just work out of the box with no `dispose` calls needed.

Reply:
I know...I was reluctant doing it from the beginning because in my implementation, I remember having issue with it, and I left this comment that made be sort of double think before doing any change. Especially since it is a bit hard to test. But I will give it a try see how it goes. 

![image](https://user-images.githubusercontent.com/27493937/120784636-968d2f00-c52c-11eb-92f8-2ab87e19af34.png)

Reply:
And as expected some of the tests failed when removing the disposed (man it's to have tests :D) I tried both with and without curly brackets. What seems to fix it, is closing the `gzipStream`  before trying to read data from the memory stream.